ga_dashboard/src/ga_dashboard/backend/ga_tools.py

```
@@ -45,20 +46,26 @@
     }
 
 
-class GA_tools():
+class GA_tools:
 
-    def __init__(self, cluster_info, fParams):
+    def __init__(self, cluster_info, fixed_params):
         self.cluster_info = cluster_info
-        self.fParams = fParams
+        self.fixed_params = fixed_params
 
     def calculate_energies(self, row):
         '''
-        Calculate the energy usaged based on the job's paramaters
+        Calculate the energy usage based on the job's parameters
         :param row: [pd.Series] one row of usage statistics, corresponding to one job
         :return: [pd.Series] the same statistics with the energies added
         '''
         ### CPU and GPU
-        partition_info = self.cluster_info['partitions'][row.PartitionX]
+        try:
+            partition_info = self.cluster_info['partitions'][row.PartitionX]
+        except KeyError as ke:
```

What's the motivation behind this exception?

---

The motivation was that I was just trying to get this script to work. Obviously we will want to make our code as robust as possible, and raise exceptions when information is missing/incomplete/corrupted

---

Comment and `TODO` added to code.

---

This could an additional thing to build: a check that the cluster_info file contains all the required field (check format ?)
And we could also do the same for the Slurm data file(s).

---

Yes, @ens-lgil. That sort of thing is definitely what we will want to check for, and make sure the code is working properly (and catching missing fields) with the unit tests.

---

That'd work. Also if this bit of code is just to make the code run for now but not a long-term test, it'd be could to highlight that with a comment so that we can keep track. I've been using the `# DEBUGONLY` tag for that to quickly identify line that can be removed, but any system would work.

---

Well, eventually we'll ideally check for missing keys every time we try to access them, so I'll leave it as-is for now.

---

Doesn't it introduce a lot of unnecessary checks? If we know we've just created the fields at a previous step, we don't need to check that it's there? Only when there is the option of a missing field it's worth checking.

---

They are not unnecessary. I added them because the code kept crashing at that point.

In any case, it's best to make our code as robust as we can, and fail early with meaningful information to the user.